clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/NonClusteredEagerEvictionScheduler.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.clustering.ejb.infinispan;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.wildfly.clustering.dispatcher.Command;
+import org.wildfly.clustering.dispatcher.CommandDispatcher;
+import org.wildfly.clustering.dispatcher.CommandDispatcherException;
+import org.wildfly.clustering.dispatcher.CommandDispatcherFactory;
+import org.wildfly.clustering.ejb.infinispan.logging.InfinispanEjbLogger;
+import org.wildfly.clustering.infinispan.spi.distribution.Locality;
+
+/**
+ * Scheduler for eager eviction of a bean.
+ * @author Paul Ferraro
+ * @author Flavia Rainone
+ */
+public class NonClusteredEagerEvictionScheduler<I, T> implements Scheduler<I>, BeanGroupEvictor<I>, Consumer<I> {
+
+    private final BeanFactory<I, T> factory;
+    private final ScheduledExecutorService executor;
+    private final Duration idleTimeout;
+    private final ExpirationTracker<I> expirationTracker;
+    private final Map<I, I> groupIdMap = new HashMap<>();
+    private volatile Future<?> evictionTask;
+
+    private final CommandDispatcher<BeanGroupEvictor<I>> dispatcher;
+
+    NonClusteredEagerEvictionScheduler(BeanFactory<I, T> factory, BeanGroupEvictor<I> evictor, ScheduledExecutorService executor, Duration idleTimeout, CommandDispatcherFactory dispatcherFactory, String dispatcherName) {
+        this.factory = factory;
+        this.executor = executor;
+        this.idleTimeout = idleTimeout;
+        this.expirationTracker = new ExpirationTracker<>(idleTimeout);
+        this.dispatcher = dispatcherFactory.createCommandDispatcher(dispatcherName + "/eviction", evictor);
```

Comment:
This scheduler no longer has any need to create a CommandDispatcher, no?

Reply:
done